From 7f7851e79f4c1d61b834b42157eac165f492b1e3 Mon Sep 17 00:00:00 2001
From: Dmitry Ornatsky <cdo021c@motorola.com>
Date: Tue, 27 Oct 2015 11:46:26 -0500
Subject: [PATCH 430/959] IKKRNBSP-3474 input: gpio-keys: log keys event in
 kernel log

In some cases the same input event can be generated by several
sources. Need to trace gpio-keys activity in kernel

Change-Id: I35ebac269fcf7c6892c132cec7d5d44a96af418e
Signed-off-by: Dmitry Ornatsky <cdo021c@motorola.com>
Reviewed-on: http://gerrit.mot.com/795131
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/input/keyboard/gpio_keys.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/keyboard/gpio_keys.c b/drivers/input/keyboard/gpio_keys.c
index 8eda2894c4b..5b8af767ddf 100644
--- a/drivers/input/keyboard/gpio_keys.c
+++ b/drivers/input/keyboard/gpio_keys.c
@@ -30,6 +30,7 @@
 #include <linux/of_gpio.h>
 #include <linux/spinlock.h>
 #include <linux/pinctrl/consumer.h>
+#include <linux/printk.h>
 
 struct gpio_button_data {
 	const struct gpio_keys_button *button;
@@ -337,6 +338,9 @@ static void gpio_keys_gpio_report_event(struct gpio_button_data *bdata)
 	} else {
 		input_event(input, type, button->code, !!state);
 	}
+		dev_info_ratelimited(&input->dev,
+			"gpio-keys report %s [%#x] type %#x state %s\n",
+			button->desc, button->code, type, state ? "On" : "Off");
 	input_sync(input);
 }
 
-- 
2.11.0

